webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PrepareTemplateVariables.scala

```diff
@@ -150,6 +152,31 @@ class PrepareTemplateVariablesImpl(
       , agentNodeConfig.config.nodeInfo.isPolicyServer
       , agentNodeConfig.config.nodeInfo.serverRoles
     )
+
+    /*
+     * Policy server certificates for the node
+     */
+    val policyServerCertificates = {
+      def getCertifiate(nodeId: NodeId): IOResult[String] = {
+        allNodeConfigs.get(nodeId) match {
+          case None    =>
+            "".succeed.tap(_ => PolicyGenerationLoggerPure.error(s"Policy server '${nodeId.value}' was not found during generation: can not provide its certificate"))
```

Comment:
is it a succeed?

Reply:
it uses to be like that because we chose to not fail policy generation for a missing/bad key or certificate. 
I think we may be able to revisit that assumption, perhaps for 7.1, if we have node by node generation by default, and after cleaning code thanks to the fact that we only have certificate for everything in 7.0.

Reply:
I added a comment